api/src/modules/materials/material.repository.ts

```diff
@@ -112,4 +113,26 @@ export class MaterialRepository extends ExtendedTreeRepository<
     // Recursively find elements and their ancestry given Ids of the subquery above
     return this.getEntityAncestry<Material>(queryBuilder, Material.name);
   }
+
+  async checkActiveMaterials(materialIds: string[]): Promise<void> {
+    const inactiveSelectedMaterials: Material[] = await this.dataSource
```

Comment:
I think this is already being handled, but note that you are using dataSource's `.getRepository()` method to get the material's repository, when you already are in the MaterialRepository class

Reply:
yep, thanks, all logic moved to service